cli/dcoscli/package/main.py

```diff
@@ -206,6 +216,171 @@ def _remove_repo(repo_name):
+
+def _resolve_local_references(package_json, package_directory):
+    """ Resolves all local refereces in package json
+    :param package_json: The package json that may contain local references
+    :type package_json: dict
+    :param package_directory: The directory of the project.
+    :type package_directory: str
+    :returns: The package json with all references resolved
+    :rtype: dict
+    """
+    bundle_schema_path = "data/schemas/bundle-schema.json"
+    bundle_schema = util.load_jsons(
+        pkg_resources.resource_string("dcoscli", bundle_schema_path).decode("utf-8"))
+
+    ref = "marathon"
+    tp = "v2AppMustacheTemplate"
+    if ref in package_json and _is_local_reference(package_json[ref][tp]):
+        location = (package_json[ref])[tp][1:]
+        if not os.path.isabs(location):
+            location = os.path.join(package_directory, location)
+
+        # convert the contents of the marathon file into base64
+        with util.open_file(location) as f:
+            contents = base64.b64encode(f.read().encode("utf-8")).decode("utf-8")
+
+        package_json[ref][tp] = contents
+
+        errs = util.validate_json(package_json, bundle_schema)
+        if errs:
+            raise DCOSException('Error validating package: '
+                                '[{}] does not conform to the specified schema'.format(location))
+
+    ref = "resource"
+    if ref in package_json and _is_local_reference(package_json[ref]):
+        location = package_json[ref][1:]
+        if not os.path.isabs(location):
+            location = os.path.join(package_directory, location)
+
+        with util.open_file(location) as f:
+            contents = util.load_json(f, True)
+
+        package_json[ref] = contents
+
+        errs = util.validate_json(package_json, bundle_schema)
+        if errs:
+            raise DCOSException('Error validating package: '
+                                '[{}] does not conform to the specified schema'.format(location))
+
+    ref = "config"
+    if ref in package_json and _is_local_reference(package_json[ref]):
+        location = package_json[ref][1:]
+        if not os.path.isabs(location):
+            location = os.path.join(package_directory, location)
+
+        with util.open_file(location) as f:
+            contents = util.load_json(f, True)
+
+        package_json[ref] = contents
+
+        errs = util.validate_json(package_json, bundle_schema)
+        if errs:
+            raise DCOSException('Error validating package: '
+                                '[{}] does not conform to the specified schema'.format(location))
+
+    return package_json
+
+
+def _is_local_reference(item):
+    """Checks if an object is a local reference
+
+   :param item: the object that may be a reference
+   :type item: object
+   :returns: true if item is a local reference else false
+   :rtype: bool
+   """
+    local_ref_pattern = re.compile("^@")
+    return isinstance(item, str) and local_ref_pattern.match(item)
```

Comment:
We don't need to use regex. You just need to know that the string to start with `@`.